pkg/granted/registry/add.go

```diff
@@ -76,84 +77,163 @@ var AddCommand = cli.Command{
+				Filename:     configFileName,
+				RequiredKeys: requiredKey,
+				Interactive:  true,
+			})
 
-		if err != nil {
-			return err
-		}
+			if err != nil {
+				return err
+			}
+			src, err := registry.AWSProfiles(ctx)
+			if err != nil {
+				return err
+			}
 
-		src, err := registry.AWSProfiles(ctx)
-		if err != nil {
-			return err
-		}
+			dst, filepath, err := loadAWSConfigFile()
+			if err != nil {
+				return err
+			}
 
-		dst, filepath, err := loadAWSConfigFile()
-		if err != nil {
-			return err
-		}
+			m := awsmerge.Merger{}
 
-		m := awsmerge.Merger{}
+			merged, err := m.WithRegistry(src, dst, awsmerge.RegistryOpts{
+				Name:                    name,
+				PrefixAllProfiles:       prefixAllProfiles,
+				PrefixDuplicateProfiles: prefixDuplicateProfiles,
+			})
+			var dpe awsmerge.DuplicateProfileError
+			if errors.As(err, &dpe) {
+				clio.Warnf(err.Error())
+
+				const (
+					DUPLICATE = "Add registry name as prefix to all duplicate profiles for this registry"
+					ABORT     = "Abort, I will manually fix this"
+				)
+
+				options := []string{DUPLICATE, ABORT}
+
+				in := survey.Select{Message: "Please select which option would you like to choose to resolve: ", Options: options}
+				var selected string
+				err = testable.AskOne(&in, &selected)
+				if err != nil {
+					return err
+				}
+
+				if selected == ABORT {
+					return fmt.Errorf("aborting sync for registry %s", name)
+				}
+
+				registryConfig.PrefixDuplicateProfiles = true
+
+				// try and merge again
+				merged, err = m.WithRegistry(src, dst, awsmerge.RegistryOpts{
+					Name:                    name,
+					PrefixAllProfiles:       prefixAllProfiles,
+					PrefixDuplicateProfiles: true,
+				})
+				if err != nil {
+					return fmt.Errorf("error after trying to merge profiles again: %w", err)
+				}
+			}
 
-		merged, err := m.WithRegistry(src, dst, awsmerge.RegistryOpts{
-			Name:                    name,
-			PrefixAllProfiles:       prefixAllProfiles,
-			PrefixDuplicateProfiles: prefixDuplicateProfiles,
-		})
-		var dpe awsmerge.DuplicateProfileError
-		if errors.As(err, &dpe) {
-			clio.Warnf(err.Error())
+			// we have verified that this registry is a valid one and sync is completed.
+			// so save the new registry to config file.
+			gConf.ProfileRegistry.Registries = append(gConf.ProfileRegistry.Registries, registryConfig)
+			err = gConf.Save()
+			if err != nil {
+				return err
+			}
 
-			const (
-				DUPLICATE = "Add registry name as prefix to all duplicate profiles for this registry"
-				ABORT     = "Abort, I will manually fix this"
-			)
+			err = merged.SaveTo(filepath)
+			if err != nil {
+				return err
+			}
 
-			options := []string{DUPLICATE, ABORT}
+			return nil
+		} else {
 
-			in := survey.Select{Message: "Please select which option would you like to choose to resolve: ", Options: options}
-			var selected string
-			err = testable.AskOne(&in, &selected)
+			registry, err := cfregistry.New(cfregistry.Opts{
```

Comment:
Why did the code in this section onwards need to be changed? (on initial glance it looks mostly the same as the current implementation, but just shifted around?)

Reply:
The only thing changed in this file was including the check at the top to handle the registry type. I'd say the diff is effected by this as it is now wrapped in a conditional